src/server/tiered_storage.cc

```diff
@@ -396,6 +396,23 @@ error_code TieredStorage::ScheduleOffload(DbIndex db_index, PrimeIterator it) {
 
   bin_record.pending_entries.insert(it->first);
 
+  PrimeTable* pt = db_slice_.GetTables(db_index).first;
+  DbSlice::Context db_context{db_index, GetCurrentTimeMs()};
+  auto erase_deleted_or_expire = [pt, db_context, this](CompactObjectView key_view) {
```

Comment:
why did you decide to delete here and not during the `PerformDeletion()` call to erase pending entries ?

Reply:
@romange actually why didn't we delete erase pending entries there initially?

Reply:
because I did not write the complete code.